uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -956,6 +956,31 @@ public function upgrade_1041() {
     return TRUE;
   }
 
+  /**
+   * Updates current case types so they have a category assigned. All case types
+   * are assigned the Workflow category by default except for the Application case
+   * type, which gets the Vacancy category.
+   *
+   * @return bool
+   */
+  public function upgrade_1042() {
+    $categoryFieldId = CRM_Core_BAO_CustomField::getCustomFieldID('category', 'case_type_category');
+    $caseTypes = civicrm_api3('CaseType', 'get', [
+      'options' => [ 'limit' => 0 ]
+    ]);
+
+    foreach ($caseTypes['values'] as $caseType) {
+      $category = $caseType['name'] === 'Application' ? 'Vacancy' : 'Workflow';
+
+      civicrm_api3('CaseType', 'create', [
+        'id' => $caseType['id'],
+        'custom_' . $categoryFieldId => $category,
```

Comment:
Very small point, but since `custom_123` array key won't change inside the loop why not define it outside of it?